### PR TITLE
Refactor circle ci config and add a job for node 14

### PR DESCRIPTION
**Pull request by gkhunger-sf**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,79 +1,105 @@
...
-  build-12:
-    environment:
-      COMMERCE_SDK_INPUT_DIR: ../test/raml/valid/
-    working_directory: ~/build_only
+# EXECUTORS
+executors:
+  node-10:
+    <<: *defaults
+    docker:
+      - image: circleci/node:10.16.3
```
Comment (bredmond-sf) [inline .circleci/config.yml:28]: Maybe a good time to upgrade? https://nodejs.org/en/about/releases/

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,79 +1,105 @@
...
+      - image: redis:5-alpine
+  node-12:
+    <<: *defaults
     docker:
       - image: circleci/node:12.16.1
       - image: redis:5-alpine
+  node-14:
+    <<: *defaults
+    docker:
+      - image: circleci/node:14
```
Comment (bredmond-sf) [inline .circleci/config.yml:38]: We should be consistent about what version we use and should specify the full version here.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,79 +1,105 @@
-jobs:
-  build:
-    environment:
-      COMMERCE_SDK_INPUT_DIR: ../test/raml/valid/
-    working_directory: ~/build_only
-    docker:
-      - image: circleci/node:10.16.3
-      - image: redis:5-alpine
+# DEFAULTS
```
Comment (bredmond-sf) [inline .circleci/config.yml:1]: Nicely done

Comment (gkhunger-sf) [reply .circleci/config.yml:1]: Thanks 🙂

**Pull request merged**